src/server.rs

```diff
@@ -301,8 +302,11 @@ impl HttpServer {
         // current thread until at least one event is received.
         // The received notifications will then populate the `events` array with
         // `event_count` elements, where 1 <= event_count <= MAX_CONNECTIONS.
-        let event_count =
-            epoll::wait(self.epoll_fd, -1, &mut events[..]).map_err(ServerError::IOError)?;
+        let event_count = match epoll::wait(self.epoll_fd, -1, &mut events[..]) {
+            Ok(event_count) => event_count,
+            Err(e) if e.kind() == std::io::ErrorKind::Interrupted => 0,
```

Comment:
nit: Add a comment which gives some insight why this handling is needed: "`epoll::wait` might be interrupted by signals and return with `libc::EINTR`. This does not break any `micro-http` functionality, and needs to return gracefully."